ProcessMaker/processmaker
# FOUR-18315 | Include Tests For Accurate Filtering and Display of Templates

# Task
Ticket: [FOUR-18315](https://processmaker.atlassian.net/browse/FOUR-18315)

This PR introduces comprehensive Cypress tests for the Screen Templates Panel in `screen-builder` and adds template-fetching functionality to `screen-builder`.

The tests check that:
- My Templates are displayed when the 'My Templates' button is clicked in the Screen Templates Panel
- Shared Templares are displayed when the 'Shared Templates' button is clicked in the Screen Templates Panel

This PR also refactors the handling of templates data, and the show/close functionality for the Screen Templates Panel.

# How to Test
1. Go to branch `task/FOUR-18315` in `screen-builder` and `processmaker`.
2. Run `npx open cypress` and `npm run dev` in your `screen-builder` terminal.
3. Run tests for `ScreenTemplateSection.spec.js` in Cypress.
	- All tests should pass.

ci:next

[Screen Builder PR](https://github.com/ProcessMaker/screen-builder/pull/1700)

# Code Review Checklist


[FOUR-18315]: https://processmaker.atlassian.net/browse/FOUR-18315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
